WBooks/AddNew/AddNewController.swift

```diff
@@ -7,23 +7,165 @@
+        setBindings()
+        setFormValidation()
+        setSubmitButton()
+        setImagePicker()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        configureNavBar()
+    }
+    
+    private func setBindings() {
+        _viewModel.bookName <~ _view.booksName.reactive.continuousTextValues.skipNil()
+        _viewModel.author <~ _view.author.reactive.continuousTextValues.skipNil()
+        _viewModel.year <~ _view.year.reactive.continuousTextValues.skipNil()
+        _viewModel.genre <~ _view.genre.reactive.continuousTextValues.skipNil()
+        _viewModel.descrpition <~ _view.bookDescription.reactive.continuousTextValues.skipNil()
+    }
+    
+    private func setFormValidation() {
+        let bookNameValidation = _view.booksName.reactive.continuousTextValues.skipNil().map { text in
+            !text.isEmpty
+        }
+        let authorValidation = _view.author.reactive.continuousTextValues.skipNil().map { text in
+            !text.isEmpty
+        }
+        let yearValidation = _view.year.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let genreValidation = _view.genre.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let descriptionValidation = _view.bookDescription.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let isButtonEnabled = Signal<Bool, NoError>.combineLatest(bookNameValidation, authorValidation, yearValidation, genreValidation, descriptionValidation).map {
+            $0 && $1 && $2 && $3 && $4
+        }
+        
+        isButtonEnabled.observeValues { [weak self] value in
+            self?.formFilled.value = value
+        }
+    }
+    
+    private func setSubmitButton() {
+        formFilled.producer.startWithValues { [weak self] formFilled in
+            if let this = self {
+                if formFilled && !this.loading {
+                    this._view.enableSubmit()
+                } else {
+                    this._view.disableSubmit()
+                }
+            }
+        }
+        _view.submitButton.reactive.controlEvents(.touchUpInside).observeValues { [weak self] _ in
+            self?.loading = true
+            self?._view.disableSubmit()
+            self?._view.disableInteractions()
+            self?.submit()
+        }
+    }
+    
+    func submit() {
+        _viewModel.addBook(onSuccess: onAddNewSuccess, onError: onError)
+    }
+    
+    func onAddNewSuccess() {
+        loading = false
+        _view.resetForm()
+        _view.enableInteractions()
+        let alert = UIAlertController(alertViewModel: ErrorAlertViewModel(title: "THANKS".localized(), message: "BOOK_ADDED".localized(), dismissButtonTitle: "ACCEPT".localized()))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func onError(error: Error) {
+        loading = true
```

Comment:
I think that if you got an error you wont be able to submit again if this is true 🤔

Reply:
That is a typo right there. It is supposed to be false. I'll change it!